drivers/include/bme680.h

```diff
@@ -211,111 +57,145 @@ typedef union {
 } bme680_intf_t;
 
 /**
- * @brief   BME680 device initialization parameters
+ * @brief   Device initialization parameters
  */
 typedef struct {
-    uint8_t ifsel;              /**< Interface selection */
-    uint8_t temp_os;            /**< Temperature oversampling */
-    uint8_t hum_os;             /**< Humidity oversampling */
-    uint8_t pres_os;            /**< Pressure oversampling */
-    uint8_t filter;             /**< IIR filter coefficient */
-    uint8_t gas_measure;        /**< Enable gas measurement */
-    uint16_t heater_dur;        /**< Heater duration in ms */
-    uint16_t heater_temp;       /**< Heater temperature in °C */
-    uint8_t power_mode;         /**< Power mode (sleep or forced) */
-    uint8_t settings;           /**< Settings used */
-    bme680_intf_t intf;         /**< Hardware interface parameters */
+    uint8_t ambient_temp;
+    uint8_t temp_os;
+    uint8_t press_os;
+    uint8_t hum_os;
+    bool meas_gas;
+    uint16_t gas_heating_time;
+    uint16_t gas_heating_temp;
+    uint8_t filter;
+    bme680_intf_t intf;
 } bme680_params_t;
 
 /**
- * @brief   BME680 device descriptor
+ * @brief   Device calibration data
  */
 typedef struct {
-    struct bme680_dev sensor; /**< Inherited device structure from vendor API */
-    bme680_intf_t intf;       /**< Device interface */
-} bme680_t;
+    uint16_t par_t1;
+    int16_t par_t2;
+    int8_t par_t3;
+    uint16_t par_h1;
+    uint16_t par_h2;
+    int8_t par_h3;
+    int8_t par_h4;
+    int8_t par_h5;
+    uint8_t par_h6;
+    int8_t par_h7; 
+    uint16_t par_p1;
+    int16_t par_p2;
+    int8_t par_p3;
+    int16_t par_p4;
+    int16_t par_p5;
+    int8_t par_p6;
+    int8_t par_p7;
+    int16_t par_p8;
+    int16_t par_p9;
+    uint8_t par_p10;
+    int8_t par_g1;
+    int16_t par_g2;
+    int8_t par_g3;
+    int8_t res_heat_range;
+} bme680_calib_t;
```

Comment:
OK, here I agree that no meaningful documentation is possible for the members. But again, due to padding memory is wasted.